components/MapView.js

```diff
@@ -446,8 +446,45 @@ class MapView extends React.Component {
   }
 
   takeSnapshot(width, height, region, callback) {
-    const finalRegion = region || this.props.region || this.props.initialRegion;
-    this._runCommand('takeSnapshot', [width, height, finalRegion, callback]);
+    return new Promise((resolve, reject) => {
+      let options;
+      if (typeof width === 'object') {
+        options = width;
+      } else {
+        options = {
```

Comment:
Can you add a deprecation warning for this case? That way we can remove the old API in the next breaking release.